src/networks.js

```diff
@@ -21,6 +21,47 @@
  * }} SocialNetwork
  */
 
+function shareOnMastodon(d) {
+  // Inspired by https://www.256kilobytes.com/content/show/4812/
+  // License: "Feel free to use and/or modify this code on your site"
+
+  const defaultInstance = (
+    localStorage['mastodon.instance'] || 'https://mastodon.social'
+  );
+
+  // eslint-disable-next-line no-alert
+  let instance = prompt(
```

Comment:
`prompt` is, well, fine, however, if the user earlier has clicked the "Prevent Site from Creating Anymore Dialogs" checkbox, this prompt will not appear. I don't think there should be a reason for the user not to be able to share something if the webpage nagged him with `prompt`s or `alert`s before.

Perhaps it would be easier to build my own popup (similar to https://github.com/Aly-ve/Mastodon-share-button, which is also too large) or a whole statically hosted mastodon share page...

Reply:
Would you like to solve this question before you consider merging?  I went for `prompt` because it seemed like it might be better than nothing, but if it's too ugly for you to want it, then we would need do some work.

That work might be tricky, because we don't know anything about what the rest of the page looks like, so any dialog would need to be customisable and have good defaults.

Reply:
A whole statically hosted mastodon share page would be great and could have features like being integrated with joinmastodon.org to get a list of instances, plus it would remove all the complexity from this PR.  But, it would be non-trivial amounts of work.

Reply:
So, I have spent a day and came up with this: https://github.com/NickKaramoff/toot. Was pretty trivial after all :)

We can generate text like we do it for WhatsApp: https://github.com/NickKaramoff/shareon/blob/bc1fa34391c9d2f99cb360587bf9322bb7cf9cef/src/networks.js#L89

And then supply it to https://toot.karamoff.dev with the `text` URL param. So, I envision something like this:

```js
const NETWORKS = {
  mastodon: {
    color: '...',
    icon: '...',
    url: (d) => `https://toot.karamoff.dev/?text=${d.title}%0D%0A${d.url}${d.text ? `%0D%0A%0D%0A${d.text}` : ''}${d.via ? `%0D%0A%0D%0A${d.via}` : ''}`,
  }
}
```

Reply:
Wow, this is really great, and solves the problem in a much better way than I was proposing.